Components/Overlay/src/OgreOverlayElement.cpp

```diff
@@ -329,10 +331,19 @@ namespace Ogre {
             return;
 
         mMaterial->load();
+
+        auto dstPass = mMaterial->getBestTechnique()->getPass(0);
+        if (dstPass->getLightingEnabled() || dstPass->getDepthCheckEnabled())
```

Comment:
What is about "ReceiveShadows"?

Reply:
I only added this recently. Not sure whether I will keep it this way.

Adding it to the check triggers with every overlay material of the sample browser. So it will likely spam all users.